.ipp files are not installed

### What happens?

Files with the `.ipp` extension aren't installed alongside the other header files. There are a few of these included by `duckdb/common/shared_ptr.hpp`, so code that depends on duckdb's C++ API fails to compile.

### To Reproduce

In a duckdb checkout, run:
```bash
GEN=ninja CMAKE_INSTALL_PREFIX=/tmp/duckdb make release
```
followed by:
```bash
cmake --install build/release --prefix /tmp/duckdb
```
Then:
```bash
echo "#include <duckdb.hpp>" | g++ -I /tmp/duckdb/include --std=c++11 -x c++ -

In file included from <stdin>:1:
In file included from /tmp/duckdb/include/duckdb.hpp:11:
In file included from /tmp/duckdb/include/duckdb/main/connection.hpp:12:
In file included from /tmp/duckdb/include/duckdb/common/serializer/buffered_file_writer.hpp:11:
In file included from /tmp/duckdb/include/duckdb/common/serializer/write_stream.hpp:11:
In file included from /tmp/duckdb/include/duckdb/catalog/catalog.hpp:11:
In file included from /tmp/duckdb/include/duckdb/catalog/catalog_entry.hpp:11:
In file included from /tmp/duckdb/include/duckdb/common/common.hpp:12:
In file included from /tmp/duckdb/include/duckdb/common/helper.hpp:12:
/tmp/duckdb/include/duckdb/common/shared_ptr.hpp:40:10: fatal error: 'duckdb/common/shared_ptr.ipp' file not found
#include "duckdb/common/shared_ptr.ipp"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```


### OS:

NixOS 23.05, macOS 14.4.1

### DuckDB Version:

0.10.3, main

### DuckDB Client:

C++

### Full Name:

Serge Monkewitz

### Affiliation:

Arista Networks

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Not applicable - the reproduction does not require a data set

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -116,7 +116,8 @@ else()
     DIRECTORY "${PROJECT_SOURCE_DIR}/src/include/duckdb"
     DESTINATION "${INSTALL_INCLUDE_DIR}"
     FILES_MATCHING
-    PATTERN "*.hpp")
+    PATTERN "*.hpp"
+    PATTERN "*.ipp")
   install(FILES "${PROJECT_SOURCE_DIR}/src/include/duckdb.hpp"
                 "${PROJECT_SOURCE_DIR}/src/include/duckdb.h"
           DESTINATION "${INSTALL_INCLUDE_DIR}")

